app/helpers/cookies.server.tsx

```diff
@@ -0,0 +1,50 @@
+import {
+  COLOR_MODE_COOKIE_NAME,
+  type ColorMode,
+  PREFERS_DARK_COOKIE_NAME,
+  THEME_COOKIE_NAME,
+  type Theme,
+} from '~/hooks/use-theme';
+
+/**
+ * Settings saved as plain cookie values used for
+ * theming, language and similar settings to be
+ * saved across visits to the app
+ */
+export type CookieSettings = {
+  theme: Theme;
+  colorMode: ColorMode;
+  prefersDark: boolean;
```

Comment:
Prefers dark means their system is set to dark mode. We store this in a cookie so that when the app loads for the first time we already know their system mode. otherwise there is a slight flicker to figure out the system mode on the frontend. And we get some errors about the ssd html not matching the frontend

Reply:
why don't we just call it systemMode and store 'dark' | 'light'? sounds more clear to me

Reply:
Yea that's more clear, changed